packages/web-api-send-notification-runner/src/tasks/notification-sender-web-api-client.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { injectable } from 'inversify';
+import { ResponseAsJSON } from 'request';
+import * as requestPromise from 'request-promise';
+import { NotificationSenderMetadata } from '../types/notification-sender-metadata';
+
+// tslint:disable: no-null-keyword no-any
+export interface ResponseWithBodyType<T = {}> extends ResponseAsJSON {
+    body: T;
+}
+
+@injectable()
+export class NotificationSenderWebAPIClient {
+    private readonly defaultRequestObject: typeof requestPromise;
+    private readonly defaultOptions: requestPromise.RequestPromiseOptions = {
+        forever: true,
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        resolveWithFullResponse: true,
+        json: true,
+    };
+
+    constructor(private readonly throwOnRequestFailure: boolean = false, httpRequest: any = requestPromise) {
+        // tslint:disable-next-line: no-unsafe-any
+        this.defaultRequestObject = httpRequest.defaults({
+            ...this.defaultOptions,
+            simple: this.throwOnRequestFailure,
+        });
+    }
+
+    public async postNotificationUrl(notificationSenderConfigData: NotificationSenderMetadata): Promise<ResponseAsJSON> {
```

Comment:
rename to sendNotification / postNotification. (you are not posting the url)

Reply:
I mean that's what we named our api "postScanUrl" in A11yServiceClient, but I can change it

Reply:
yes. we should rename that too. it is not posting scan url. it is notifying that the scan is completed.